samples/net/echo_server/src/echo-server.c

```diff
@@ -70,21 +102,17 @@ struct net_pkt *build_reply_pkt(const char *name,
 
 	recv_len = net_pkt_get_len(pkt);
 
-	tmp = pkt->frags;
-
 	/* If we have link layer headers, then get rid of them here. */
 	if (recv_len != net_pkt_appdatalen(pkt)) {
 		/* First fragment will contain IP header so move the data
 		 * down in order to get rid of it.
 		 */
-		header_len = net_pkt_appdata(pkt) - tmp->data;
+		header_len = net_pkt_ip_hdr_len(pkt);
+		header_len += net_pkt_ipv6_ext_len(pkt);
+		header_len += proto_len;
 
-		NET_ASSERT(header_len < CONFIG_NET_BUF_DATA_SIZE);
-
-		/* After this pull, the tmp->data points directly to application
-		 * data.
-		 */
-		net_buf_pull(tmp, header_len);
+		NET_DBG("pull of header %d", header_len);
```

Comment:
This will print too much information as debugging is enabled by default. Thus this print can be removed.